util/test/sub_test

```diff
@@ -1122,21 +1117,29 @@ for testname in testsrc:
             numTrials = 1
 
     # Get list of test specific compiler options
+    # Default to [' ']
+    compoptslist = list(' ')
+
     chpldoc_opts_filename = test_filename + chpldocsuffix
     if test_is_chpldoc and os.access(chpldoc_opts_filename, os.R_OK):
         compoptslist = ReadFileWithComments(chpldoc_opts_filename, False)
         if not compoptslist:
             sys.stdout.write('[Warning: ignoring an empty chpldocopts file %s]\n' %
                              (test_filename+compoptssuffix))
-            compoptslists = list(' ')
     elif os.access(test_filename+compoptssuffix, os.R_OK):
         compoptslist = ReadFileWithComments(test_filename+compoptssuffix, False)
         if not compoptslist:
             # cf. for execoptslist no warning is issued
             sys.stdout.write('[Warning: ignoring an empty compopts file %s]\n'%(test_filename+compoptssuffix))
-            compoptslist = list(' ')
-    else:
-        compoptslist = list(' ')
+
+    # Merge global compopts list with local compopts.
+    # Use the "product" of the two if they are both provided.
+    usecompoptslist = [ ]
+    for dir_compopts in directoryCompopts:
+      for file_compopts in compoptslist:
+        useopt = [dir_compopts, file_compopts]
+        usecompoptslist += [' '.join(useopt)]
```

Comment:
You could use [itertools.product](https://docs.python.org/2/library/itertools.html#itertools.product) here:

```python
compoptslist = []
for dir_compopts, file_compopts in itertools.product(directoryCompopts, compoptsList):
    compoptslist.append(' '.join([dir_compopts, file_compopts]))
```